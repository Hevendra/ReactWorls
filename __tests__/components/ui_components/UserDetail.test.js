import React from 'react';
import {shallow} from 'enzyme';
import {UserDetail} from '../../../src/components/ui_components/UserDetail';
import renderer from 'react-test-renderer';

describe ('Component UserDetail', () => {
  const testData = {
    avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/mrmoiree/128.jpg',
    first_name: 'Hevendra',
    last_name: 'Jadaun',
  };
  const actionDelete = () => {};

  it ('should render without throwing an error', () => {
    expect (
      shallow (
        <UserDetail data={testData} handleDelete={actionDelete} />
      ).exists (<div className="card-body" />)
    ).toBe (true);
  });

  it ('should render avatar', () => {
    expect (
      shallow (
        <UserDetail data={testData} handleDelete={actionDelete} />
      ).find ('#image_avatar').length
    ).toEqual (1);
  });

  it ('Component Snapshot test', () => {
    const tree = renderer
      .create (<UserDetail data={testData} handleDelete={actionDelete} />)
      .toJSON ();
    expect (tree).toMatchSnapshot ();
  });
});
