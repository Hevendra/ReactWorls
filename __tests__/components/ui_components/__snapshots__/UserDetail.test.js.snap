// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UserDetail Component Snapshot test 1`] = `
<div
  className="card-body"
>
  <div
    className="wrapper"
  >
    <img
      alt="CZ"
      className="image--cover"
      id="image_avatar"
      src=""
    />
  </div>
  <p
    className="display-name"
  >
    Hevendra Jadaun
     
  </p>
  <p
    className="button"
    onClick={[Function]}
  >
    <span>
      Delete 
    </span>
  </p>
</div>
`;
